openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -100,6 +100,17 @@ TCN_IMPLEMENT_CALL(jlong, SSLContext, make)(TCN_STDARGS, jint protocol, jint mod
 
     UNREFERENCED(o);
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
+    // TODO this is very hacky as io.netty.handler.ssl.OpenSsl#doesSupportProtocol also uses this method to test for supported protocols. Furthermore
```

Comment:
is the intention to fix/improve this in Netty as things have changed in tcnative? Most of this code was carried over from apache's tcnative so we can trim/modify it if it makes things cleaner/easier.